apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/ServerInfoControllerTest.java

```diff
@@ -0,0 +1,203 @@
+        MockHttpResponse response = new MockHttpResponse();
+        dispatcher.invoke(request, response);
+        // check
+        assertEquals(200, response.getStatus());
+        String respondJson = response.getContentAsString();
+        BlockchainStatusDto dtoResult = mapper.readValue(respondJson, new TypeReference<>(){});
+        assertNotNull(dtoResult.application);
+        assertNotNull(dtoResult.version);
+        // verify
+        verify(serverInfoService, times(1)).getBlockchainStatus();
+    }
+
+    @Test
+    void blockchainConstants_SUCCESS() throws URISyntaxException, IOException {
+        // prepare data
+        BlockchainConstantsDto dto = new BlockchainConstantsDto("123-567", "234-567", 100L);
+        doReturn(dto).when(serverInfoService).getBlockchainConstants();
+        // init mocks
+        ServerInfoController controller = new ServerInfoController(serverInfoService);
+        dispatcher.getRegistry().addSingletonResource(controller);
+        // call
+        String uri = blockchainConstantsUri;
+        MockHttpRequest request = MockHttpRequest.get(uri);
+        MockHttpResponse response = new MockHttpResponse();
+        dispatcher.invoke(request, response);
+        // check
+        assertEquals(200, response.getStatus());
+        String respondJson = response.getContentAsString();
+        BlockchainConstantsDto dtoResult = mapper.readValue(respondJson, new TypeReference<>(){});
+        assertNotNull(dtoResult.genesisAccountId);
+        assertEquals(100L, dtoResult.epochBeginning);
+        // verify
+        verify(serverInfoService, times(1)).getBlockchainConstants();
+    }
+
+    @Test
+    void blockchainState_SUCCESS() throws URISyntaxException, IOException {
+        // prepare data
+        BlockchainStateDto dto = new BlockchainStateDto(
+            new BlockchainStatusDto("1.48.1", "1.2", 123, BlockchainState.UP_TO_DATE.toString()));
+        doReturn(dto).when(serverInfoService).getBlockchainState(true);
+        // init mocks
+        ServerInfoController controller = new ServerInfoController(serverInfoService);
+        dispatcher.getRegistry().addSingletonResource(controller);
+
+        // call
+        String uri = blockchainStateUri + "?includeCounts=" + Boolean.TRUE + "&adminPassword=1";
+        MockHttpRequest request = MockHttpRequest.get(uri);
+        MockHttpResponse response = new MockHttpResponse();
+        dispatcher.invoke(request, response);
+        // check
+        assertEquals(200, response.getStatus());
+        String respondJson = response.getContentAsString();
+        BlockchainStateDto dtoResult = mapper.readValue(respondJson, new TypeReference<>(){});
+        assertNotNull(dtoResult.application);
+        assertNotNull(dtoResult.version);
+        assertEquals(123, dtoResult.time);
+        assertEquals(BlockchainState.UP_TO_DATE.toString(), dtoResult.blockchainState);
+        // verify
+        verify(serverInfoService, times(1)).getBlockchainState(true);
```

Comment:
you can to not verify calls for mocked methods, since its already verified by mockito and when mocked method was not invoked exception will be raised.

Reply:
Incorrect assumption, that doesn't work that way. No verification is done without using `verify()` method.